### ISSUE TITLE
Exception executing the SlackAPIPostOperator

### ISSUE THREAD

**Issue by r39132** [apache/airflow#689]
I'm on 1.6.1+ (i.e. HEAD master as of yesterday)

I'm trying to use the following SlackAPIPostOperator

```
post_completion_to_slack_job = SlackAPIPostOperator(
    task_id='post_completion_to_slack_job',
    token='XXXXXXX-redacted-XXXXXXXX',
    channel='#eng-notifications',    
    text='test',
    icon_url='http://airbnb.io/img/projects/airflow3.png',
    dag=dag)
```

Here is what it looks like in my DAG: 

<img width="628" alt="screenshot 2015-11-24 23 28 54" src="https://cloud.githubusercontent.com/assets/581734/11390908/2a4d10be-9303-11e5-9ae7-5c78c933b2b7.png">

But I get the following exception when running it with `airflow test`

```
Traceback (most recent call last):
  File "/usr/local/bin/airflow", line 5, in <module>
    pkg_resources.run_script('airflow==1.6.1', 'airflow')
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 528, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 1394, in run_script
    execfile(script_filename, namespace, namespace)
  File "/usr/local/lib/python2.7/dist-packages/airflow-1.6.1-py2.7.egg/EGG-INFO/scripts/airflow", line 17, in <module>
    args.func(args)
  File "/usr/local/lib/python2.7/dist-packages/airflow-1.6.1-py2.7.egg/airflow/bin/cli.py", line 287, in test
    ti.run(force=True, ignore_dependencies=True, test_mode=True)
  File "/usr/local/lib/python2.7/dist-packages/airflow-1.6.1-py2.7.egg/airflow/models.py", line 984, in run
    result = task_copy.execute(context=context)
  File "/usr/local/lib/python2.7/dist-packages/airflow-1.6.1-py2.7.egg/airflow/operators/slack_operator.py", line 50, in execute
    sc.api_call(self.method, **self.params)
  File "/usr/local/lib/python2.7/dist-packages/slackclient/_client.py", line 21, in api_call
    return self.server.api_call(method, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/slackclient/_server.py", line 133, in api_call
    reply = self.api_requester.do(self.token, method, kwargs)
  File "/usr/local/lib/python2.7/dist-packages/slackclient/_slackrequest.py", line 21, in do
    return urlopen(url, post_data.encode('utf-8'))
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 171, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 494, in open
    response = self._open(req, data)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 517, in _open
    'unknown_open', req)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 466, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python2.7/dist-packages/future/backports/urllib/request.py", line 1353, in unknown_open
    raise URLError('unknown url type: %s' % type)
future.backports.urllib.error.URLError: <urlopen error unknown url type: https>
```

**Comment by r39132** [comment]
Everyone, please +1 one this so that we can get some momentum behind this : https://github.com/slackhq/python-slackclient/pull/48

**Comment by syvineckruyk** [comment]
@r39132 I have yet to run in to this error ... is only when using airflow test that you encounter it ?

**Comment by mistercrunch** [comment]
Might be specific to the python version you are using. It has to do with the compatibility of url related calls in 2.7 and 3+

**Comment by tom-dyar** [comment]
I ran into it using 2.7.6 -- anyone have a workaround?

**Comment by mistercrunch** [comment]
+1 this here:
https://github.com/slackhq/python-slackclient/issues/49

I fixed the issue on their repo and they have merged it, they just haven't released it to pypi yet

**Comment by dgies** [comment]
This was fixed with the release of SlackClient 0.17
https://pypi.python.org/pypi/slackclient/0.17
